plotly/tests/test_core/test_utils/test_utils.py

```diff
@@ -50,3 +52,98 @@ def test_node_generator(self):
         ]
         for i, item in enumerate(node_generator(node0)):
             self.assertEqual(item, expected_node_path_tuples[i])
+
+
+class TestMemoizeDecorator(TestCase):
+
+    # In Python 2.x, globals should be module-scoped. By defining and
+    # instantiating a class, we *access* the global first before attempting
+    # to update a value. I.e., you *cannot* simply mutate the global value
+    # on it's own.
+    class Namespace(object):
+        pass
+
+    def test_memoize(self):
+        name_space = self.Namespace()
+        name_space.call_count = 0
+
+        @memoize()
+        def add(a, b):
+            name_space.call_count += 1
+            return a + b
+
+        tests = [[(1, 1), 2], [(2, 3), 5], [(3, -3), 0]]
+
+        self.assertEqual(name_space.call_count, 0)
+        for i, (inputs, result) in enumerate(tests, 1):
+            for _ in range(10):
```

Comment:
Out of curiosity, is `_` the canonical (or a typical) variable used for loops where the looping variable is not used inside?